src/core/electrostatics/p3m_gpu_cuda.cu

```diff
@@ -521,18 +525,16 @@ void assign_forces(P3MGpuData const &params,
 
   /* Switch for assignment templates, the shared version only is faster for cao
    * > 2 */
-  auto const data_length =
-      3u *
-      static_cast<std::size_t>(parts_per_block *
-                               static_cast<unsigned int>(cao)) *
-      sizeof(float);
-  switch (cao) {
+  auto const data_length = std::size_t(3u) *
+                           static_cast<std::size_t>(parts_per_block) *
+                           static_cast<std::size_t>(cao) * sizeof(float);
+  switch (params.cao) {
```

Comment:
To avoid implicit conversion compiler diagnostics. At line 503, the local `cao` is no longer a signed int: `auto const cao = static_cast<unsigned int>(params.cao);`